### PR TITLE
test: use fail-rs for some tests.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

**Comment by siddontang** [comment]
Cool work!

**Comment by ngaut** [comment]
Feels good to see more clean up.

**Comment by siddontang** [comment]
PTAL @AndreMouche

File: tests/failpoints.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #![feature(fnbox)]
 #![feature(slice_patterns)]
 #![feature(box_syntax)]
+#![feature(test)]
```
Comment (siddontang) [inline tests/failpoints.rs]: do we need this feature here?

Comment (disksing) [reply tests/failpoints.rs]: this PR includes `mod storage`, and there are some bench test inside `mod storage`.

**Pull request merged**